pkg/rbd/nodeserver.go

```diff
@@ -538,6 +539,48 @@ func (ns *NodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstag
 	return &csi.NodeUnstageVolumeResponse{}, nil
 }
 
+func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
+
+	volumeID := req.GetVolumeId()
+	if volumeID == "" {
+		return nil, status.Error(codes.InvalidArgument, "NodeExpandVolume volume ID must be provided")
+	}
+	volumePath := req.GetVolumePath()
+	if volumePath == "" {
+		return nil, status.Error(codes.InvalidArgument, "NodeExpandVolume volume path must be provided")
+	}
+
+	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: mount.NewOsExec()}
+	volumePath = volumePath + "/" + req.GetVolumeId()
+	devicePath, err := getDevicePath(ctx, diskMounter, volumePath)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	// TODO check size and return success or error
+	resizer := resizefs.NewResizeFs(diskMounter)
+	ok, err := resizer.Resize(devicePath, volumePath)
+	if !ok {
+		return nil, fmt.Errorf("rbd: resize failed on path %s, error: %v", req.GetVolumePath(), err)
+	}
+	return &csi.NodeExpandVolumeResponse{
+		CapacityBytes: req.GetCapacityRange().GetRequiredBytes(),
+	}, nil
+}
+
+func getDevicePath(ctx context.Context, mnt *mount.SafeFormatAndMount, stagingTargetPath string) (string, error) {
```

Comment:
(nit) `stagingTargetPath` can be renamed as this can be either the staging or the publish path. Hence keeping this as `volumePath` or some equivalent would avoid assumptions when reading the code.